### ISSUE TITLE
try catch finally

### ISSUE THREAD

**Issue by loberyyq** [nus-cs2030/2223-s1#302]
Is this section very important? coz in our normal coding I didn't use that, when is it die die that we must use this framework

**Comment by YuZizhen** [comment]
I think when we dont know whether our code or the user's input will generate an error, we will need to use the try and catch to filter out potential errors.

**Comment by dohaduong** [comment]
I think it's not too important for cs2030 as we should focus on effect-free methods

**Comment by dohaduong** [comment]
I think it's not too important for cs2030 as we are focused on effect-free methods

**Comment by dohaduong** [comment]
I think it's not too important for cs2030 cuz we focus more on effect-free methods

**Comment by ZYN2122** [comment]
I think generally the prof asks us not to use these as well

**Comment by YeoWeiHan123** [comment]
I don't think this is tested for exam. 
This is try catch is usually for exception and make client understand the error better
example 
```public class Main {
  public static void main(String[ ] args) {
    int[] myNumbers = {1, 2, 3};
    System.out.println(myNumbers[10]); // error!
  }
}
```
would give an error
`Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 10
        at Main.main(Main.java:4) `
however it would be better to include a catch and print a custom error so client can better understand the error
```public class Main {
  public static void main(String[ ] args) {
    try {
      int[] myNumbers = {1, 2, 3};
      System.out.println(myNumbers[10]);
    } catch (Exception e) {
      System.out.println("Error: number must be between myNumber range.");
    }
  }
}
```
would result in
`Error: number must be between myNumber range.`

**Comment by KEEWEIKUANG** [comment]
Prof went through this lecture saying that it is not that important in cs2030 yup

**Comment by Yu-Haikuo** [comment]
Yes it is not important. I remember Prof also said this portion will not be tested in the final exam.

**Comment by giorgiocana** [comment]
It is very useful in practical usage. During my internship I was using try catches quite a lot. But I don't think it will be tested for this course since prof said it was an optional lecture.

**Comment by chiralcentre** [comment]
I will agree with @giorgiocana try catches are more important in a practical setting but it should not be tested for finals, since no input validation is required.

**Comment by wjiayis** [comment]
Ya I agree, Prof Henry mentioned that the focus is on business logic of the code and rather than try-catch.

**Comment by Grauda123** [comment]
it can help with reducing if else statements too

**Comment by mohddanesh** [comment]
it is not important for final
